### PR TITLE
issue #174

### PR DESCRIPTION
**Pull request by pythonguru101**

### CODE REVIEW

**Comment by gijzelaerr** [comment]
hi! thanks for your pull request. I'm traveling at the moment but will review this when I'm home.

**Comment by pythonguru101** [comment]
Hi Gijs
Okay, Please review it and I will wait for your feedback
Looking forward to working on the next issue asap 😊
Have a great weekend!
Best Regards

Sent from Mail for Windows 10

File: eduvpn/steps/totp_enroll.py [hunk 1]
```diff
@@ -49,7 +51,8 @@ def _make_qr(builder, oauth, meta, config_dict, lets_connect, secret=None):
 
 
 # ui thread
-def _parse_user_input(builder, oauth, meta, config_dict, lets_connect, secret=None):
+def _parse_user_input(builder, oauth, meta, config_dict, lets_connect, secret=None)
```
Comment (gijzelaerr) [inline eduvpn/steps/totp_enroll.py]: there is a `:` missing here.

File: eduvpn/steps/browser.py [hunk 2]
```diff
@@ -69,13 +71,15 @@ def _phase1_background(meta, dialog, verifier, builder, force_token_refresh, let
 
 
 def _phase1_callback(meta, port, code_verifier, oauth, auth_url, dialog, builder, state, lets_connect):
+    # type : (Metadata, port, code_verifier, oauth, auth_url, Gtk.Dialog, Gtk.Builder, state, bool) -> None
     thread_helper(lambda: _phase2_background(meta=meta, port=port, oauth=oauth, code_verifier=code_verifier,
```
Comment (gijzelaerr) [inline eduvpn/steps/browser.py]: Test suite fails on this point now.

**Comment by gijzelaerr** [comment]
We use travis for running our test suite, and as you can see it fails for all builds:

https://travis-ci.org/eduvpn/python-eduvpn-client/builds/534216504

I'm not sure why, but the github travis integration seems to be not working properly, normally there should be a link on this page directing us to this travis page, but maybe @fkooman disabled that for security reasons.

**Comment by fkooman** [comment]
Not sure what happened... I removed all Travis-CI integration for the server components a looong time ago as I don't need it and was just chasing after Travis-CI bugs and incompatibilities that don't occur on "real" systems.

By the way: Fedora 27 and 28 are EOL, so it makes (more) sense to build on Fedora 29 and 30 instead.

**Comment by gijzelaerr** [comment]
yep, there is an open issue for that https://github.com/eduvpn/python-eduvpn-client/issues/172

**Comment by pythonguru101** [comment]
I’ve just fixed it
Please check it 

Sent from Mail for Windows 10

**Comment by pythonguru101** [comment]
Not sure what happened... I removed all Travis-CI integration for the server components a looong time ago as I don't need it and was just chasing after Travis-CI bugs and incompatibilities that don't occur on "real" systems.
By the way: Fedora 27 and 28 are EOL, so it makes (more) sense to build on Fedora 29 and 30 instead.

**Comment by pythonguru101** [comment]
Please try to build again, it will be a success.

**Comment by gijzelaerr** [comment]
@fkooman can you turn the travis integration on again?

**Comment by gijzelaerr** [comment]
@pythonguru101 you only fixed the missing `:`, the other issue still an issue. Please run the test suite before requesting a review.

**Comment by pythonguru101** [comment]
@pythonguru101 you only fixed the missing :, the other issue still an issue. Please run the test suite before requesting a review.

**Comment by fkooman** [comment]
> @fkooman can you turn the travis integration on again?

Not sure, I am no longer an admin of the eduVPN organization... @mkoelewijn @efef

**Comment by pythonguru101** [comment]
Before commit, I will double check the patches on my side

**Comment by gijzelaerr** [comment]
made a new issue for that https://github.com/eduvpn/python-eduvpn-client/issues/179

**Comment by pythonguru101** [comment]
yeah, will review new issue

**Comment by gijzelaerr** [comment]
The test suite still fails on the same location

File: eduvpn/__init__.py [hunk 3]
```diff
@@ -5,7 +5,4 @@
 
 import pkg_resources
 
-try:
-    __version__ = pkg_resources.require("eduvpn")[0].version # type: str
-except pkg_resources.DistributionNotFound:
-    __version__ = "0.0dev" # type: str
+__version__ = pkg_resources.require("eduvpn")[0].version # type: str
```
Comment (gijzelaerr) [inline eduvpn/__init__.py]: this is not right, that try except is there for a reason, if one uses the module but the eduvpn is not completely installed it doesn't know the version, so in that case it returns `0.0dev`. If mypy gives an error here then that is new (a new version of mypy) and that error should be muted.

File: eduvpn/steps/browser.py [hunk 4]
```diff
@@ -116,7 +116,7 @@ def _phase2_background(meta, port, oauth, code_verifier, auth_url, dialog, build
             logger.error("received from state, expected: {}, received: {}".format(state, other_state))
             raise Exception("oauth state has been tampered with")
         logger.info("setting oauth token for metadata")
-        meta.token = oauth.fetch_token(meta.token_endpoint, code=code, code_verifier=code_verifier, client_id=oauth.client_id, include_client_id=True)
```
Comment (gijzelaerr) [inline eduvpn/steps/browser.py]: are you sure you can just remove these arguments? they seem quite important to me.

**Comment by pythonguru101** [comment]
@gijzelaerr requested changes on this pull request.



In eduvpn/steps/browser.py:
 @@ -116,7 +116,7 @@ def _phase2_background(meta, port, oauth, code_verifier, auth_url, dialog, build
             logger.error("received from state, expected: {}, received: {}".format(state, other_state))
             raise Exception("oauth state has been tampered with")
         logger.info("setting oauth token for metadata")
-        meta.token = oauth.fetch_token(meta.token_endpoint, code=code, code_verifier=code_verifier, client_id=oauth.client_id, include_client_id=True)

are you sure you can just remove these arguments? they seem quite important to me.

**Comment by pythonguru101** [comment]
Currently there are 3 arguments in fetch_token() in utils.py
And it returns empty value now

**Comment by gijzelaerr** [comment]
i'm not sure what you are saying, but the tests are failing for only some platforms now:

https://travis-ci.org/eduvpn/python-eduvpn-client/builds/535448841

probably there is a mismatch between the API of multiple oauth libraries on the various supported platforms.

**Comment by gijzelaerr** [comment]
do you know how to run the test suite with all the docker containers? that can be helpful to verify the test suite runs on all platforms.

**Comment by pythonguru101** [comment]
When I built the Python/Django project, I have worked with docker container
So I know some docker commands that handle the docker containers
For example
docker ps -a
docker exec -it CONTAINER_ID /bin/bash
....
At first, I'd like to run the test suit without any pull request locally

**Comment by gijzelaerr** [comment]
If you look in the `Makefile` you will find a couple of development helper targets. `make dockers` will build containers for all supported platforms and run the test suite inside them.

**Comment by pythonguru101** [comment]
Yes, I know
If you can provide me a detailed guide, please send me it
Currently, I am trying to be familiar with this project
And did you check the that functions' arguments
I think that the function's arguments do not correspond to it's prototype
What do you think about it?

**Pull request closed without merging**